### PR TITLE
Use current replicatoin factor to expand partition if it is larger than the config

### PR DESCRIPTION
**Pull request by lindong28**

### CODE REVIEW

**Comment by lindong28** [comment]
@smccauliff Can you please review this patch?

File: src/main/java/com/linkedin/kmf/services/MultiClusterTopicManagementService.java [hunk 1]
```diff
@@ -242,16 +242,17 @@ void maybeReassignPartitionAndElectLeader() throws Exception {
           throw new IllegalStateException("Topic " + _topic + " does not exist in cluster " + _zkConnect);
 
         int currentReplicationFactor = getReplicationFactor(partitionInfoList);
+        int expectedReplicationFactor = Math.max(currentReplicationFactor, _replicationFactor);
 
         if (_replicationFactor < currentReplicationFactor)
-          throw new RuntimeException(String.format("Configured replication factor %d "
+          LOG.debug(String.format("Configured replication factor %d "
```
Comment (smccauliff) [inline src/main/java/com/linkedin/kmf/services/MultiClusterTopicManagementService.java]: For log message  use {} rather than String.format

**Comment by smccauliff** [comment]
Other than that one comment this LGTM.

File: src/main/java/com/linkedin/kmf/services/MultiClusterTopicManagementService.java (hunk 1 above)
Comment (lindong28) [inline src/main/java/com/linkedin/kmf/services/MultiClusterTopicManagementService.java]: Thanks @smccauliff. It is fixed now.

Comment (lindong28) [inline src/main/java/com/linkedin/kmf/services/MultiClusterTopicManagementService.java]: I messed up the commit. Will update soon.

**Pull request merged**